### DISCUSSION TITLE
Map Command+S shortcut on every page where applicable

### DISCUSSION THREAD

**Discussion by rijkvanzanten** [directus/directus#7046] [Feature Requests]
### Discussed in https://github.com/directus/directus/discussions/6581

<div type='discussions-op-text'>

<sup>Originally posted by **RSickenberg** June 30, 2021</sup>
Hello, 

I realized there is a shortcut for saving with cmd+s but not everywhere, is it normal?

Regards;
</div>

**Comment by vanling** [comment]
Dropping my 2 cents here, found this via issue: The place i notice cmd+s is not working is when focussed in a WYSIWYG field.

**Reply by rijkvanzanten** [reply]
Yes! Noticed that too. I think it's because it's inside the `contenteditable` context which then triggers the browser differently. Or it's because TinyMCE is swallowing all events somehow 🤔

**Reply by vanling** [reply]
Yep, TinyMCE is swallowing xD 

Running in console
```
tinymce.activeEditor.shortcuts.remove('meta+s', '', '')
tinymce.activeEditor.shortcuts.add('meta+s', '', (event)=>{console.log(event)})
// cmd+s returns undefined in console
```

what ever i try it cancels bubbling up

**Reply by vanling** [reply]
Is there an save event i can call ? 

```
tinymce.activeEditor.shortcuts.remove('meta+s', '', '')
tinymce.activeEditor.shortcuts.add('meta+s', '', ()=>{CALL_DIRECTUS_SAVE_ACTION()})
```


The vue setup is a bit tricky, init here is a string/object and unable to run setup() there. Im going to try something like this later :) 
`<Editor v-on:init="replaceSaveShortcut" />`
```
const replaceSaveShortcut = (editor)=>{
    editor.shortcuts.remove('meta+s');
    editor.shortcuts.add('meta+s', '', () => {console.log('overwritten shortcut')});
}
```

edit: found something like this `if (hasEdits.value) saveAndStay();`
edit2:  🎉https://github.com/directus/directus/blob/df054f294de148669e71ab80e046ea83c8325175/app/src/interfaces/input-rich-text-html/input-rich-text-html.vue#L474-L488

**Reply by rijkvanzanten** [reply]
There's currently no way for the individual interfaces to call the parent's page "save" function. I'm wondering if we can somehow prevent tinymce from capturing that event in the first place, and instead bubble it up so it's handled by the regular event listener?

**Reply by vanling** [reply]
Unfortunately `editor.shortcuts.remove('meta+s');` will remove the existing one but still prevents it from bubbling up. 

Maybe we can (re)trigger the shortcut  from here some how 
https://github.com/directus/directus/blob/d6846d74ebb36a7765a6a0a1960363875b16b3c0/app/src/composables/use-shortcut.ts

fabricate keyboard event and use `callHandlers('')`

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
